File: app/src/main/java/com/mailerdaemon/app/ImpContacts/ImpContactActivity.java
package com.mailerdaemon.app.ImpContacts;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.LinearLayout;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.mailerdaemon.app.R;

import java.util.Objects;

import Utils.ContactFunction;
import butterknife.BindView;
import butterknife.ButterKnife;

public class ImpContactActivity extends AppCompatActivity implements ContactFunction {

    @BindView(R.id.bt_faculty)
    LinearLayout bt_faculty;
    @BindView(R.id.bt_admin)
    LinearLayout bt_admin;
    @BindView(R.id.bt_hostel)
    LinearLayout bt_hostel;
    @BindView(R.id.bt_senate)
    LinearLayout bt_senate;
    @BindView(R.id.more)
    LinearLayout bt_more;


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_imp_contacts);
        ButterKnife.bind(this);
        Toolbar toolbar=findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setTitle("Contacts");

        bt_faculty.setOnClickListener(v -> {
            String[] tabs = new String[]{"All", "ACH","AGL","AGP","AM","AP","CHE","CIV","CSE","ECE","EE","ESE","FME","HSS","ME","MECH","MME","MS","PE"};
            String[] pages = new String[]{"faculty_all","ACH","AGL","AGP","AM","AP","CHE","CIV","CSE","ECE","EE","ESE","FME","HSS","ME","MECH","MME","MS","PE"};
            openDetail(tabs, pages);
        });

        bt_admin.setOnClickListener(v -> {
            String[] tabs = new String[]{"Deans", "Associate Deans", "HOD", "HOC"};
            String[] pages = new String[]{"deans", "associate_deans", "hod", "hoc"};
            openDetail(tabs, pages);
        });
        bt_hostel.setOnClickListener(v -> openWebView("file:///android_asset/warden.html"));
        bt_senate.setOnClickListener(v->{

        });
        bt_more.setOnClickListener(v->{

        });
    }

    public void openDetail(String[] tabs, String[] pages) {
        ContactFragmentViewPager fragment = new ContactFragmentViewPager();
        Bundle bundle = new Bundle();
        bundle.putStringArray("tabs", tabs);
        bundle.putStringArray("pages", pages);
        fragment.setArguments(bundle);
        getSupportFragmentManager().beginTransaction().add(R.id.fragment_container, fragment, null).commit();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        if (item.getItemId() == android.R.id.home)
            onBackPressed();
        return true;

    }

    @Override
    public void onBackPressed() {
        if(getSupportFragmentManager().getFragments().isEmpty())
        super.onBackPressed();
        else getSupportFragmentManager().beginTransaction().remove(getSupportFragmentManager().getFragments().get(0)).commit();
    }

    void openWebView(String url){
        HtmlFragment fragment= new HtmlFragment();
        Bundle bundle=new Bundle();
        bundle.putString("url",url);
        fragment.setArguments(bundle);
        getSupportFragmentManager().beginTransaction().add(R.id.fragment_container,fragment, null).addToBackStack(null).commit();
    }

    @Override
    public void makeCall(String num) {
        if(!num.trim().equals("0")){
            Intent intent = new Intent(Intent.ACTION_DIAL, Uri.parse("tel:" + num.trim()));
            startActivity(intent);}
        else{
            Toast.makeText(this,"Sorry number not available",Toast.LENGTH_LONG).show();
        }
    }

    @Override
    public void sendMail(String s) {
        if(!s.trim().isEmpty()) {
            Intent send = new Intent(Intent.ACTION_SENDTO);
            String uriText = "mailto:" + Uri.encode(s.trim()) +
                    "?subject=" + Uri.encode("Subject") +
                    "&body=" + Uri.encode("the body of the message");
            Uri uri = Uri.parse(uriText);

            send.setData(uri);
            startActivity(Intent.createChooser(send, "Send mail..."));
        }
        else{
            Toast.makeText(this,"Sorry email not available",Toast.LENGTH_LONG).show();
        }
    }
}

File: app/src/main/java/com/mailerdaemon/app/impContacts/ImpContactActivity.kt
package com.mailerdaemon.app.impContacts

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.mailerdaemon.app.R
import com.mailerdaemon.app.WARDEN_LINK
import com.mailerdaemon.app.toast
import com.mailerdaemon.app.utils.ChromeTab
import com.mailerdaemon.app.utils.ContactFunction
import kotlinx.android.synthetic.main.activity_imp_contacts.*

class ImpContactActivity : AppCompatActivity(), ContactFunction {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_imp_contacts)
        setSupportActionBar(toolbar)
        supportActionBar?.let {
            it.setDisplayHomeAsUpEnabled(true)
            it.title = resources.getString(R.string.title_activity_imp_contacts)
        }
        bt_faculty.setOnClickListener { openDetail(resources.getStringArray(R.array.tabs),
                resources.getStringArray(R.array.pages)) }
        bt_admin.setOnClickListener {
            val tabs = arrayOf("Deans", "Associate Deans", "HOD", "HOC")
            val pages = arrayOf("deans", "associate_deans", "hod", "hoc")
            openDetail(tabs, pages)
        }
        bt_hostel.setOnClickListener {
            val tab = ChromeTab(this)
            tab.openTab(WARDEN_LINK)
        }
    }

    private fun openDetail(tabs: Array<String>, pages: Array<String>) {
        val fragment = ContactFragmentViewPager()
        val bundle = Bundle()
        bundle.putStringArray("tabs", tabs)
        bundle.putStringArray("pages", pages)
        fragment.arguments = bundle
        supportFragmentManager.beginTransaction().add(R.id.fragment_container, fragment, null).commit()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId === android.R.id.home) onBackPressed()
        return true
    }

    override fun onBackPressed() {
        if (supportFragmentManager.fragments.isEmpty()) super.onBackPressed()
        else supportFragmentManager.beginTransaction().remove(supportFragmentManager.fragments[0]).commit()
    }

    fun openWebView(url: String) {
        val fragment = HtmlFragment()
        val bundle = Bundle()
        bundle.putString("url", url)
        fragment.arguments = bundle
        supportFragmentManager.beginTransaction().add(R.id.fragment_container, fragment, null)
                .addToBackStack(null).commit()
    }

    override fun makeCall(num: String) {
        if (num.trim() != "0") {
            val intent = Intent(Intent.ACTION_DIAL, Uri.parse("tel:" + num.trim()))
            startActivity(intent)
        } else this.toast(resources.getString(R.string.number_not_available))
    }

    override fun sendMail(s: String) {
        if (s.trim().isNotEmpty()) {
            val send = Intent(Intent.ACTION_SENDTO)
            val uriText = "mailto:${Uri.encode(s.trim())}" +
                    "?subject=${Uri.encode("Subject")}" +
                    "&body=${Uri.encode("the body of the message")}"
            send.data = Uri.parse(uriText)
            startActivity(Intent.createChooser(send, "Send mail..."))
        } else this.toast(resources.getString(R.string.email_not_available))
    }
}
